Restructure readme

luyong6 opened the pull request:
This time I made some big changes to README. I hope this is more clear. 
I added the link pointing to "doc" folder and mentioned "QuickStart". Please modify the words and sentences for my inappropriate expression. 

Signed by Lu Yong (luyong@cn.ibm.com)

jsvogt commented on README.md:
Bullets are for lists, not for headings. Use ### instead.

jsvogt commented on README.md:
```diff
@@ -14,64 +15,83 @@ Please see here for more details:
-Please see [Bitstream_flashing.md](hardware/doc/Bitstream_flashing.md) for instructions on how to update the FPGA bitstream.
+Developing an FPGA accelerated application on SNAP takes following steps: 
+* **Preparation**: Decide the software function to be moved to FPGA. This function, usually computation intensive, is named as "action" in the following description. Carefully read the **Dependencies** in following section. Choose an FPGA card. Install tools, download packages and set environmental variables.
 
```
I believe we shouldn't "enforce" these steps and annoy the experts. We can propose this as one way, but experienced FPGA developers may simply jump right into an existing example and tweak it to their needs.

luyong6 commented:
Hi @jsvogt will you also merge this into #710? 
I modified the * to ###, added the declaration of supporting 8k5 card, and simplified some wordings in "Preparation" step and so on. 
Please refine the wordings from your side.
